### PR TITLE
Add storybook entry for search bar

### PR DESCRIPTION
**Pull request by hschallhorn**
Bumps #15646

### Description
Creates a storybook entry for search bar. Better documentation of the component. Remove unused props.

### Acceptance Criteria
- [x] Stories exist for all components mentioned in the existing Caseflow Style Guide
  - [x] Search
- [x] Story _AT LEAST_ includes same [documentation](https://storybook.js.org/docs/react/writing-docs/introduction) as the style guide (look for ways to add more!)
- [x] Story _AT LEAST_ includes same examples ([stories](https://storybook.js.org/docs/react/writing-stories/introduction#how-to-write-stories)) as the style guide (look for ways to add more!)
- [x] Add proptype documentation via js docstrings to components that are missing them ([these will be automatically pulled by storybook](https://storybook.js.org/docs/react/writing-docs/doc-blocks#argstable))
- [x] Add [controls](https://storybook.js.org/docs/react/essentials/controls) for args that will change the display of the component

### Testing Plan
1. Run `make storybook` and navigate to http://localhost:6006/?path=/docs/commons-components-searchbar--big-story

### User Facing Changes

 BEFORE|AFTER|AFTER 2
 ---|---|---
Screen Shot 2020-11-30 at 3 14 54 PM|Screen Shot 2020-11-30 at 4 19 57 PM|Screen Shot 2020-11-30 at 4 20 16 PM

### Code Documentation Updates
- [x] Add or update code comments at the top of the class, module, and/or component.

### Storybook Story
*For Frontend (Presentationa) Components*
* [x] Add a [Storybook](https://github.com/department-of-veterans-affairs/caseflow/wiki/Documenting-React-Components-with-Storybook) file alongside the component file (e.g. create `MyComponent.stories.js` alongside `MyComponent.jsx`)
* [x] Give it a title that reflects the component's location within the overall Caseflow hierarchy
* [x] Write a separate story (within the same file) for each discrete variation of the component

### CODE REVIEW

File: client/app/components/SearchBar.jsx [hunk 1]
```diff
@@ -12,9 +12,6 @@ export default class SearchBar extends React.Component {
     this.props.onChange(event.target.value);
   }
 
-  // A "search" event occurs when a user finishes typing
-  // a search query. This is 500ms after the last character
-  // typed or when focus is lost.
```
Comment (hschallhorn) [inline client/app/components/SearchBar.jsx:17]: moved here: https://github.com/department-of-veterans-affairs/caseflow/pull/15655/files#diff-3551d664508e1037d9b1a42197915e2ac1bbd5f4089b2b82c200c71d3db8b7f9R199

File: client/app/components/SearchBar.jsx [hunk 2]
```diff
@@ -68,7 +65,7 @@ export default class SearchBar extends React.Component {
 
   clearInput = () => this.input.value = '';
 
-  getInternalField = (spinnerColor = COLORS.WHITE) => {
+  getInternalField = (spinnerColor) => {
```
Comment (hschallhorn) [inline client/app/components/SearchBar.jsx:68]: moved here: https://github.com/department-of-veterans-affairs/caseflow/pull/15655/files#diff-3551d664508e1037d9b1a42197915e2ac1bbd5f4089b2b82c200c71d3db8b7f9R165

File: client/app/components/SearchBar.jsx [hunk 3]
```diff
@@ -164,23 +161,91 @@ export default class SearchBar extends React.Component {
...
   size: PropTypes.string,
+
+  /**
+   * Callback fired when value is changed
+   *
+   * @param {string} value The current value of the component
+   */
   onChange: PropTypes.func,
   onKeyPress: PropTypes.func,
-  onClick: PropTypes.func,
```
Comment (hschallhorn) [inline client/app/components/SearchBar.jsx:173]: Was unused

File: client/app/components/SearchBar.jsx [hunk 4]
```diff
@@ -164,23 +161,91 @@ export default class SearchBar extends React.Component {
...
+  /**
+   * Whether or not a search is being performed. Will disable the search button and show a loading icon if true
+   */
   loading: PropTypes.bool,
+
+  /**
+   * The value of the `input` element
+   */
   value: PropTypes.string,
-  analyticsCategory: PropTypes.string,
```
Comment (hschallhorn) [inline client/app/components/SearchBar.jsx:179]: Was unused

File: client/app/reader/DocumentListHeader.jsx [hunk 5]
```diff
@@ -43,7 +43,6 @@ class DocumentListHeader extends React.Component {
             placeholder="Type to search..."
             value={props.docFilterCriteria.searchQuery}
             size="small"
-            analyticsCategory="Claims Folder"
```
Comment (hschallhorn) [inline client/app/reader/DocumentListHeader.jsx:46]: Was unused

File: client/app/components/SearchBar.jsx (hunk 4 above)
Comment (jcq) [reply client/app/components/SearchBar.jsx:179]: Love that we're clearing out unused props and other vars!

File: client/app/components/SearchBar.jsx [hunk 6]
```diff
@@ -164,23 +161,91 @@ export default class SearchBar extends React.Component {
   }
 }
 
+SearchBar.defaultProps = {
+  spinnerColor: COLORS.WHITE,
+};
+
 SearchBar.propTypes = {
+
+  /**
```
Comment (jcq) [inline client/app/components/SearchBar.jsx:170]: Seeing jsDoc comments for props warms my heart

File: client/app/components/SearchBar.stories.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+import SearchBar from './SearchBar';
+
+const handleSearchClick = (setLoadingFn) => {
+  setLoadingFn(true);
+
+  setTimeout(() => setLoadingFn(false), 2000);
+};
+
+const Template = (args) => <span className={args.parentClassName} ><SearchBar {...args} /></span>;
```
Comment (jcq) [inline client/app/components/SearchBar.stories.js:11]: Interesting need for pulling out the wrapper class

Was going to suggest destructuring the args to make it more explicitly clear that `parentClassName` doesn't actually go to `SearchBar`:

```jsx
const Template = ({parentClassName, ...searchBarArgs}) => (
<span className={args.parentClassName}>
  <SearchBar {...searchBarArgs} />
</span>
)
```
...but I'm not sure it's necessary

File: client/app/components/SearchBar.stories.mdx [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+
+## Loading
+
+When waiting for async results, it's best practices to display to the user that their request is processing.
+
+<Story story={SearchBarStories.Loading} />
+
+## Search Ahead
+
+For any search ahead search bars, the class `cf-search-ahead-parent` must be applied to the parent element.
```
Comment (jcq) [inline client/app/components/SearchBar.stories.mdx:72]: 👍  on moving this text here

**Review by jcq — APPROVED** [review]
Looks great!

I'll be honest and say that I'm not entirely sure what the behavior/functionality is supposed to be with the searchAhead variants, but yours works just like the existing Style Guide version, so that's good enough for now!

File: client/app/components/SearchBar.stories.js (hunk 7 above)
Comment (hschallhorn) [reply client/app/components/SearchBar.stories.js:11]: Love the suggestion!

**Pull request merged**